language-service/src/services/yamlTraversal.ts

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+import * as Parser from "../parser/jsonParser";
+import { YAMLDocument } from "../parser/yamlParser";
+import { PromiseConstructor, Thenable } from "vscode-json-languageservice";
+import { TextDocument, Position } from "vscode-languageserver-types";
+import { YamlNodeInfo } from "../yamlLanguageService";
+
+export class YAMLTraversal {
+
+    private promise: PromiseConstructor;
+
+    constructor(promiseConstructor: PromiseConstructor) {
+        this.promise = promiseConstructor || Promise;
+    }
+
+    public findNodes(document: TextDocument, yamlDocument: YAMLDocument, key: string): Thenable<YamlNodeInfo[]> {
+        if(!document){
+            this.promise.resolve([]);
+        }
+
+        const jsonDocument = yamlDocument.documents.length > 0 ? yamlDocument.documents[0] : null;
+        if(jsonDocument === null){
+            return this.promise.resolve([]);
+        }
+
+        let nodes: YamlNodeInfo[] = [];
+        jsonDocument.visit((node => {
+            const propertyNode = node as Parser.PropertyASTNode;
+            if (propertyNode.key && propertyNode.key.value === key && propertyNode.value.start !== propertyNode.value.end) {
```

Comment:
Why is this excluding properties that don't have values?

Reply:
It was to prevent this from returning incomplete stuff. Like if you're in the middle of writing:
- task: 
I guess its fine to return if it doesn't have a value and I'll filter out the value-less ones at a different level

Reply:
scratch that. this is checking key.value, not the value of the property. Basically is the key valid.

Reply:
The last condition is checking the length of the value, not the key.

Reply:
got it. I'll let it through but then filter them out in the yaml editor because it doesn't make sense to allow codelens commands until the task name and version is defined.